crates/uv-extract/src/stream.rs

```diff
@@ -154,15 +154,19 @@ async fn untar_in<'a>(
     Ok(())
 }
 
-/// Unzip a `.tar.gz` archive into the target directory, without requiring `Seek`.
+/// Unzip a `.tar.*` archive into the target directory, without requiring `Seek`.
 ///
 /// This is useful for unpacking files as they're being downloaded.
-pub async fn untar_gz<R: tokio::io::AsyncRead + Unpin>(
+pub(crate) async fn untar_compression<
+    R: tokio::io::AsyncRead + Unpin,
+    Decoder: tokio::io::AsyncRead + Unpin,
+>(
     reader: R,
+    decoder: impl Fn(tokio::io::BufReader<R>) -> Decoder,
```

Comment:
Can you just make this fn take the `BufReader`? And move the first few lines out to the caller?